demo/admin/src/products/categories/AssignedProductsGrid.tsx

```diff
@@ -0,0 +1,330 @@
+import { gql, useApolloClient, useQuery } from "@apollo/client";
+import {
+    CrudVisibility,
+    DataGridToolbar,
+    GridCellContent,
+    GridColDef,
+    GridColumnsButton,
+    GridFilterButton,
+    muiGridFilterToGql,
+    muiGridSortToGql,
+    ToolbarActions,
+    ToolbarFillSpace,
+    ToolbarItem,
+    useBufferedRowCount,
+    useDataGridRemote,
+    usePersistentColumnState,
+} from "@comet/admin";
+import { StateFilled } from "@comet/admin-icons";
+import { useTheme } from "@mui/material";
+import { DataGridPro, GridFilterInputSingleSelect, GridFilterInputValue, GridRenderCellParams, GridToolbarQuickFilter } from "@mui/x-data-grid-pro";
+import { GQLProductFilter } from "@src/graphql.generated";
+import { ProductsGridPreviewAction } from "@src/products/ProductsGridPreviewAction";
+import * as React from "react";
+import { FormattedMessage, FormattedNumber, useIntl } from "react-intl";
+
+import {
+    GQLProductGridRelationsQuery,
+    GQLProductGridRelationsQueryVariables,
+    GQLProductsListManualFragment,
+    GQLProductsListQuery,
+    GQLProductsListQueryVariables,
+    GQLUpdateProductStatusMutation,
+    GQLUpdateProductStatusMutationVariables,
+} from "./AssignedProductsGrid.generated";
+
+function ProductsGridToolbar({ toolbarAction }: { toolbarAction?: React.ReactNode }) {
+    return (
+        <DataGridToolbar>
+            <ToolbarItem>
+                <GridToolbarQuickFilter />
+            </ToolbarItem>
+            <ToolbarItem>
+                <GridFilterButton />
+            </ToolbarItem>
+            <ToolbarFillSpace />
+            <ToolbarItem>
+                <GridColumnsButton />
+            </ToolbarItem>
+            <ToolbarItem>{toolbarAction && <ToolbarActions>{toolbarAction}</ToolbarActions>}</ToolbarItem>
```

Comment:
Everything on the right side of `ToolbarFillSpace` should be wrapped in a single `ToolbarActions` instead of multiple `ToolbarItem` components. 

```suggestion
            <ToolbarActions>
                <GridColumnsButton />
                {toolbarAction}
            </ToolbarActions>
```

@johnnyomair @thomasdax98 maybe we should rework the toolbar components slightly so this differentiation between left and right content is not required to achieve the design defined by UX. Almost everyone does this wrong 😅

Reply:
https://vivid-planet.atlassian.net/browse/COM-1271

FYI @thomasdax98